ckanext/canada/tests/test_ati.py

```diff
@@ -1,32 +1,33 @@
 # -*- coding: UTF-8 -*-
-from nose.tools import assert_equal, assert_raises
+from nose.tools import assert_raises
 from ckanapi import LocalCKAN, ValidationError
 
 from ckan.tests.helpers import FunctionalTestBase
 from ckanext.canada.tests.factories import CanadaOrganization as Organization
 
 from ckanext.recombinant.tables import get_chromo
 
+
 class TestAti(FunctionalTestBase):
     def setup(self):
         super(TestAti, self).setup()
         org = Organization()
-        lc = LocalCKAN()
-        lc.action.recombinant_create(dataset_type='ati', owner_org=org['name'])
-        rval = lc.action.recombinant_show(dataset_type='ati', owner_org=org['name'])
+        self.lc = LocalCKAN()
```

Comment:
LocalCKAN is essentially free. There's no need to share it between tests (more sharing between tests is more potential for state to leak and then we might accidentally have tests dependent on run order)

Reply:
@wardi The `setup` method is run before every test in the class, so would `self.lc = LocalCKAN()` just create a new instance of `LocalCKAN` at the beginning of each test?

Or is there a better way to clear that variable to prevent state leakage? Such as in the `teardown` method?

Reply:
you're right, this is the other setup method than the one I was thinking of. Still there's nothing valuable in these LocalCKAN instances and no time is spent creating them so the extra complexity of keeping them on the instance isn't really worthwhile for the simple cases IMO. When setup is creating a bunch of LocalCKAN instances I think it's okay to pass them on `self` because it saves lines of code in the tests, on the other hand using fixtures would be even cleaner, right?

Reply:
@wardi yeah, fixtures would be best, but as these ones are not pytest yet we cannot use fixtures to my knowledge?

I can change the `self.lc = LocalCKAN()` stuff to fixtures when modifying them for the `canada-v2.9` branch.

Reply:
sure, that's fine.